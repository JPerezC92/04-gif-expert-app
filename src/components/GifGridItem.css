.gif__item {
  align-content: center;
  display: flex;
  flex-direction: column;
  min-width: 15em;
  margin: 1em 0;
  box-shadow: 0 0 7px 3px rgb(87, 76, 185);
}

.gif__image {
  min-width: 17em;
  max-width: 25em;
  min-height: 10em;
  max-height: 25em;
}

.gif__title {
  text-align: center;
  font-size: 1.2em;
}
