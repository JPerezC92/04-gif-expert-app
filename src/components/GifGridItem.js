import React from "react";
import PropTypes from "prop-types";
import "./GifGridItem.css";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="gif__item animate__animated animate__fadeIn animate__slow">
      <img className="gif__image" src={url} alt={title} />
      <h3 className="gif__title">{title}</h3>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
