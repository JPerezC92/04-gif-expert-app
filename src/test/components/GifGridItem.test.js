import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en el archivo GifGridItem", () => {
  const title = "one punch GIF by Adult Swim";
  const url =
    "https://media3.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?câ€¦79ada6yamuui69bx0aw98l98cmx34tfeggl0dxngpdpk1a&rid=giphy.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de comparar correctamente con la imagen", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p").text().trim();

    expect(p).toBe(title);
  });

  test("Debe de tener la imagen igual al url y alt de los props", () => {
    const { src, alt } = wrapper.find("img").props();

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de tener animate__fadeIn", () => {
    const { className } = wrapper.find("div").props();

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
