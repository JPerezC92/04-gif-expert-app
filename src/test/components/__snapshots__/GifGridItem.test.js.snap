// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GifGridItem Debe de comparar correctamente con la imagen 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__slow"
>
  <img
    alt="one punch GIF by Adult Swim"
    src="https://media3.giphy.com/media/l3vRlt8kty8KKeHni/giphy.gif?câ€¦79ada6yamuui69bx0aw98l98cmx34tfeggl0dxngpdpk1a&rid=giphy.gif"
  />
  <p>
    one punch GIF by Adult Swim
  </p>
</div>
`;
